### DISCUSSION TITLE
brew cleanup removes ~/Library/Caches/Homebrew/bootsnap every time

### DISCUSSION THREAD

**Discussion by luckman212** [Homebrew/discussions#4994] [Everyday usage]
### Output of `brew config`

```shell
$ brew config
HOMEBREW_VERSION: 4.2.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ff899e5b3ff6731208564ef502db237e2627e8ba
Last commit: 21 hours ago
Core tap JSON: 19 Dec 13:42 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_AUTO_UPDATE_SECS: 14400
HOMEBREW_AUTOREMOVE: set
HOMEBREW_BAT: set
HOMEBREW_BOOTSNAP: set
HOMEBREW_CASK_OPTS: []
HOMEBREW_CLEANUP_MAX_AGE_DAYS: 30
HOMEBREW_EDITOR: editor_subl
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_GITHUB_PACKAGES_TOKEN: set
HOMEBREW_GITHUB_PACKAGES_USER: luckman212
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_NO_INSECURE_REDIRECT: set
HOMEBREW_TEMP: /private/tmp/brew
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.43.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
$ brew dr
Your system is ready to brew.
```

### Description of issue

Since some recent version (4.2?) I get this on **every** run of `brew cleanup`

```
$ brew cleanup
Removing: /Users/luke/Library/Caches/Homebrew/bootsnap... (513 files, 3.8MB)
==> This operation has freed approximately 3.8MB of disk space.
```

I don't think this is expected/intended.

**Comment by Bo98** [comment]
There isn't really any logic to it at the moment: https://github.com/Homebrew/brew/blob/38c5e52a6d243b5bf87792a88c082cb2be7e68d3/Library/Homebrew/cleanup.rb#L509-L514

So yeah it currently always behaves like that for those who have `HOMEBREW_BOOTSNAP` set.

Bootnsap itself doesn't clean old cache entries, so it's useful to do something so it doesn't get too bloated: https://github.com/Homebrew/brew/blob/38c5e52a6d243b5bf87792a88c082cb2be7e68d3/Library/Homebrew/cleanup.rb#L509-L514.

**Comment by luckman212** [comment]
Ok gotcha. Maybe related? [Should `HOMEBREW_BOOTSNAP` even be enabled?](https://github.com/orgs/Homebrew/discussions/4629)
